### PR TITLE
Add glb-parser callbacks

### PR DESCRIPTION
**Pull request by slimbuck**
This PR adds support for load-time callbacks when loading gltf container assets and also documents the recently added texture handler options.

New API:
```javascript
pc.callbacks.ProcessAnimationCallback: function (animTrack, extras);
pc.callbacks.ProcessMaterialCallback: function (material, extras);
```

An example loading a container asset with process callbacks:
```javascript
var file = {
    filename: filename,
    url: url
};

var options = {
    processAnimation: function (animTrack, extras) {
        console.log('animation=' + animation + ' extras=' + extras);
    },
    processMaterial: function (material, extras) {
        console.log('material=' + material + ' extras=' + extras);
    }
};

var asset = new pc.Asset(filename, 'container', file, null, options);
asset.once('load', function (loadedAsset) {
    // handle loaded asset
});
asset.once('error', function (err) {
    // handle error case
});
this.app.assets.add(asset);
this.app.assets.load(asset);

```

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by willeastcott** [comment]
Can you add some example code to the PR description illustrating how you use these callbacks?

File: src/resources/texture.js [hunk 1]
```diff
@@ -136,6 +136,13 @@ Object.assign(pc, function () {
         texture._levelsUpdated = texture._cubemap ? [[true, true, true, true, true, true]] : [true];
     };
 
+    /**
+     * @class
+     * @name pc.TextureHandlerOptions
+     * @description The supported {@link pc.Asset} options when loading texture assets.
+     * @property {boolean} [crossOrigin] - For browser-supported image formats only, enable cross origin.
+     */
```
Comment (willeastcott) [inline src/resources/texture.js:144]: Where's the constructor? Have you tried compiling Typescript that uses this? I'm guessing the compiler will complain that a vanilla object can't be cast to a `pc.TextureHandlerOptions`.

File: src/asset/asset.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ Object.assign(pc, function () {
      * var asset = new pc.Asset("a texture", "texture", {
      *     url: "http://example.com/my/assets/here/texture.png"
      * });
-     * @param {object} [options] - a JSON object containing load-time options specific to the asset.
+     * @param {(pc.TextureHandlerOptions|pc.ContainerHandlerOptions)} [options] - The asset handler options.
```
Comment (willeastcott) [inline src/asset/asset.js:64]: Why not just keep this as a generic `object`?

File: src/resources/texture.js (hunk 1 above)
Comment (slimbuck) [reply src/resources/texture.js]: removed.

File: src/asset/asset.js (hunk 2 above)
Comment (slimbuck) [reply src/asset/asset.js]: done

**Comment by Maksims** [comment]
Also, there are different extensions and data on nodes.
Is it possible to subscribe to specific extensions parsing?

**Comment by slimbuck** [comment]
Hi @Maksims,

That could be very useful. What info would you expect to receive in the extensions callback?

File: src/resources/parser/glb-parser.js [hunk 3]
```diff
@@ -1172,12 +1180,12 @@ Object.assign(pc, function () {
     };
 
     // create engine resources from the downloaded GLB data
-    var createResources = function (device, gltf, buffers, textures, callback) {
+    var createResources = function (device, gltf, buffers, textures, options, callback) {
```
Comment (ellt92) [inline src/resources/parser/glb-parser.js:1183]: Could we also include a `processGlobalExtras` option as some glb assets contain extras data inside the gltf object itself?

Comment (slimbuck) [reply src/resources/parser/glb-parser.js:1183]: Consider it done!

**Comment by slimbuck** [comment]
Closing in favour of post-rollup PR.

**Pull request closed without merging**